### PR TITLE
#166730023 Ensure Eslint and Hound-ci are properly setup

### PR DESCRIPTION
**Pull request by nkpremices**
#### What does this PR do?
Have ```Hound-ci``` connected to the repo and activated. 
Have``` Eslint properly``` setup

#### Description of Task to be completed?
* Activate authors heaven on ```Hound-ci```
* Enable some ```Eslint``` rules and add additional rules
* Add a script to fix ```Eslint``` errors in the ```package.json``` file

#### How should this be manually tested?
After cloning the repo, install all the dependencies then create a new file and test all the new rules by typing some lines of code.

### CODE REVIEW

File: src/api/models/token.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const token = (sequelize, DataTypes) => {
```
Comment (MCFrank16) [inline src/api/models/token.js:1]: @nkpremices I don't think we do still need this token model as for now on we are going to use Redis.

File: src/api/controllers/password.reset.js [hunk 2]
```diff
@@ -55,25 +56,23 @@ export default class PasswordReset {
...
-        }
+    const user = await User.update({
+      password: hashedPassword
+    },
+    {
+      where: {
+        email: userEmail
       }
-    );
+    });
```
[the remark this answers is not in the document]
Comment (MCFrank16) [reply src/api/controllers/password.reset.js]: ``"function-paren-newline": ["error", "never"]`` that would be a rule to remove this hound hounding you

File: src/helpers/constants/status.codes.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export default {
```
Comment (MCFrank16) [inline src/helpers/constants/status.codes.js:1]: we really needed this man ``Genie Congolais``

File: src/api/models/token.js (hunk 1 above)
Comment (alainburindi) [reply src/api/models/token.js:1]: you are the one to do it @MCFrank16

**Review by alainburindi — APPROVED** [review]

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      {
+        newIsCap: true,
+        capIsNew: true,
+        properties: true
+      }
+    ],
+    'consistent-return': 0,
+    'no-param-reassign': 0,
+    'comma-dangle': 0,
+    curly: ['error', 'multi-line'],
```
Comment (MemunaHaruna) [inline .eslintrc.js:24]: @nkpremices there's another `curly` rule on line 52. Let's remove one of them.

File: .gitignore [hunk 5]
```diff
@@ -43,4 +43,7 @@ package-lock.json
 .env
 
 # vscode config file
-.vscode
\ No newline at end of file
+.vscode
+
+# Eslint cache 
+.eslintcache
```
Comment (MemunaHaruna) [inline .gitignore:49]: @nkpremices just a nitpick: This red symbol means you should add a new line to the end of the file. It's showing in several places, so just take a look at the PR and add a newline wherever this symbol is showing.

**Review by MemunaHaruna — APPROVED** [review]

File: .eslintrc.js (hunk 4 above)
Comment (premlogic) [reply .eslintrc.js:24]: Noted. let me remove one of them

File: .gitignore (hunk 5 above)
Comment (premlogic) [reply .gitignore:49]: I'm on it. Thank you

**Pull request merged**